tensorrt_llm/_torch/auto_deploy/shim/ad_executor.py

```diff
@@ -206,6 +207,10 @@ def _prepare_inputs(
 
             # get cache indices
             cache_indices = kv_cache_manager.get_cache_indices(request)
+            # TODO: is this the right place to do this?
+            page_size = self.cache_seq_interface.info.page_size
+            num_pages = end_compute // page_size + int(end_compute % page_size > 0)
+            cache_indices = cache_indices[:num_pages]
```

Comment:
NOTE: I would love to use just use utilities from the `kv_cache_manager` or `request` to correctly compute this